modules/RTC/TPCUtils.js

```diff
@@ -339,21 +339,41 @@ export class TPCUtils {
 
     /**
      * Replaces the existing track on a RTCRtpSender with the given track.
+     *
      * @param {JitsiLocalTrack} oldTrack - existing track on the sender that needs to be removed.
      * @param {JitsiLocalTrack} newTrack - new track that needs to be added to the sender.
-     * @returns {Promise<void>} - resolved when done.
+     * @returns {Promise<RTCRtpTransceiver>} - resolved with the associated transceiver when done, rejected otherwise.
      */
     replaceTrack(oldTrack, newTrack) {
         const mediaType = newTrack?.getType() ?? oldTrack?.getType();
-        const transceiver = this.findTransceiver(mediaType, oldTrack);
         const track = newTrack?.getTrack() ?? null;
+        let transceiver;
+
+        // If old track exists, replace the track on the corresponding sender.
+        if (oldTrack) {
+            transceiver = this.pc.peerconnection.getTransceivers()?.find(t => t.sender?.track === oldTrack.getTrack());
```

Comment:
how about putting source names as `mid` and using mlines as slots for sending local tracks that can be re-used? wouldn't that make finding the transceiver easier?

Reply:
when I'm looking at how many `?.` there are, it seems there's so many things that can go wrong there

Reply:
No we don't want to hardcode mids to the source names since we won't be able recycle them for other sources. The mid order cannot be changed at all, we can only append extra mids to the current ones. The idea is so to use the first transceiver (created by the jvb ssrcs in the remote offer) for the first local source and then create a new transceiver when a new local source is added.

Reply:
How is it a problem to create dedicated mline for each local track? It actually makes it easier to recycle the lines and to do track replacement (because you act on specific mline tied to the track). The dedicated mlines can be recycled and the only waste is that they would be inactive during the time there's no local track sent on that slot. To me it seems more brittle approach and error prone trying to rely on the ordering, but if that's easier then it's fine I guess.